extensions/msal-node-extensions/src/lock/CrossPlatformLock.ts

```diff
@@ -62,14 +63,20 @@ export class CrossPlatformLock {
      */
     public async unlock(): Promise<void> {
         try {
-            // delete lock file
-            await fs.unlink(this.lockFilePath);
-            await this.lockFileHandle.close();
+            if(this.lockFileHandle){
+                // if we have a file handle to the .lockfile, delete lock file
+                await fs.unlink(this.lockFilePath);
+                await this.lockFileHandle.close();
+                this.logger.info("lockfile deleted");
+            } else {
+                this.logger.warning("lockfile handle does not exist, so lockfile could not be deleted");
+            }
         } catch (err) {
             if (err.code == Constants.ENOENT_ERROR) {
-                this.logger.warning("Tried to unlock but Lockfile does not exist");
+                this.logger.warning("Tried to unlock but lockfile does not exist");
```

Comment:
Difference is that .lockfile missing vs unable to unlock?

Reply:
Yes. We try to delete the .lockfile. If it fails because it's not there, then it means that the file is unlocked, but we log a warning message. If it fails for whatever other reason, file still might be locked and therefore we throw.